File: dailyquest_web/src/main/java/dailyquest/user/repository/UserRepository.java
package dailyquest.user.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import dailyquest.user.entity.UserInfo;

@Repository
public interface UserRepository extends JpaRepository<UserInfo, Long> {

    UserInfo findByOauth2Id(String userId);
    boolean existsByNickname(String nickname);
}

File: user/src/main/kotlin/dailyquest/user/repository/UserRepository.kt
package dailyquest.user.repository

import dailyquest.user.entity.User
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository: JpaRepository<User, Long> {
    fun findByOauth2Id(userId: String): User?
    fun existsByNickname(nickname: String): Boolean
    fun findAllByIdIn(userIds: List<Long>, pageable: Pageable): Page<User>
}